core/variant/array.cpp

```diff
@@ -207,6 +207,19 @@ void Array::insert(int p_pos, const Variant &p_value) {
 	_p->array.insert(p_pos, p_value);
 }
 
+void Array::fill(const Variant &p_value) {
+	ERR_FAIL_COND(!_p->typed.validate(p_value, "fill"));
+	if (p_value.get_type() < Variant::DICTIONARY) {
+		for (int i = 0; i < _p->array.size(); i++) {
+			set(i, p_value);
+		}
+	} else {
+		for (int i = 0; i < _p->array.size(); i++) {
+			set(i, p_value.duplicate(true));
```

Comment:
For reference, there are some limitations/bugs with `duplicate(true)` as well #37162, so it may not work as expected for `Resource`s I presume. I don't know whether those issues are fixed in 4.0, though.